Tests/NIOHTTPCompressionTests/HTTPRequestDecompressorTest.swift

```diff
@@ -120,14 +112,10 @@ class HTTPRequestDecompressorTest: XCTestCase {
                 try channel.writeInbound(HTTPServerRequestPart.head(.init(version: .init(major: 1, minor: 1), method: .POST, uri: "https://nio.swift.org/test", headers: headers)))
             )
 
-            do {
-                try channel.writeInbound(HTTPServerRequestPart.body(compressed))
-            } catch let error as NIOHTTPDecompression.DecompressionError {
-                switch error {
-                case .limit:
-                    // ok
-                    break
-                default:
+            XCTAssertThrowsError(try channel.writeInbound(HTTPServerRequestPart.body(compressed))) { error in
+               if case .some(.limit) = error as? NIOHTTPDecompression.DecompressionError {
+                    // Okay
+                } else {
```

Comment:
And similarly let's totally revert this change, as the assertion isn't valid here either.